curp/src/server/curp_node.rs

```diff
@@ -490,49 +498,92 @@ impl<C: 'static + Command> CurpNode<C> {
                     }
                 }
             };
-            tokio::time::sleep(retry_timeout).await;
+            sleep(retry_timeout).await;
         }
     }
 
-    /// Sync task is responsible for replicating log entries
-    async fn sync_task(
-        curp: Arc<RawCurp<C>>,
-        connects: HashMap<ServerId, Arc<impl ConnectApi>>,
-        mut sync_rx: mpsc::UnboundedReceiver<LogIndex>,
+    /// invoke `send_log_to_follower` for every follower
+    fn bcast_append_entries(
+        curp: &Arc<RawCurp<C>>,
+        connects: &HashMap<ServerId, Arc<impl ConnectApi>>,
+        batch: Vec<LogIndex>,
```

Comment:
If the batch's contents are always the continuous number, can we do it better? For instance the first log index and the length.

Reply:
Fixed